performance-test/src/test/java/org/opensearch/dataprepper/test/data/generation/UserAgentTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.test.data.generation;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+class UserAgentTest {
```

Comment:
Would adding a test using a regex pattern make the format of the user agent string clearer?

Reply:
I thought about that, but I was trying to avoid depending on `ua_parser` library in here.

Reply:
@KarstenSchnitter , I added my own regex rather than use `ua_parser`. Please see the latest commit.

Reply:
The own regex was what I was asking for. Thanks for adding it.